docs/building_kilo.md

```diff
@@ -0,0 +1,89 @@
+# Build and Test Kilo
+
+Clone the Repository and `cd` into it.
+```shell
+git clone https://github.com/squat/kilo.git
+cd kilo
+```
+
+## Build
+
+The binaries are compiled in a docker container, so make sure `docker` is installed and the daemon is running.
+
+`make`, `curl`, `jq` and `go` need to be installed.
+The latter not for building, but formating.
+
+### Compile Binaries
+
+To compile the binaries `kg` and `kgctl` run:
+```shell
+make
+```
+On linux with amd64, the binaries will be stored in `/bin/linux/amd64/`.
+
+You can build the binaries for a different architecture with:
+```shell
+ARCH=<arm|arm64|amd64> make
+```
+
+To build `kg` for darwin or windows, run:
+```shell
+OS=<windows|darwin> make
+```
+## Test
+
+For unit tests, run:
+```shell
+make unit
+```
+
+To lint, run:
+```shell
+make lint
+```
+
+To execute basic end to end tests, run:
+```shell
+make e2e
+```
+__Note__: The end to end test are unreliable, so if they fail try running them again.
+
+Or simply run all test with:
+```shell
+make test
+```
+
+## Build and Push the Container Images
+
+If you want to build containers for a different processor architecture then your computer's, run:
+```shell
+docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+```
+
+To upload test containers of Kilo to your docker hub repository, run:
+```shell
+export IMAGE=<docker hub user name>
```

Comment:
.../kilo

Reply:
We probably want to add the `make` command for uploading the image too, right? like

```shell
IMAGE=<docker hub user name> make push
```
Otherwise it might be confusing since this command does not actually upload anything

Reply:
Ah, or maybe this is just setup for the final `make manifest`?  Maybe we can clarify this

Reply:
yes, the later. Yes it is a bit unclear